pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -371,45 +370,46 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		allNodeNames = append(allNodeNames, name)
 	}
 
-	var lock sync.Mutex
-	var firstError error
-
-	topologyMaps := newTopologyPairsMaps()
-
-	appendTopologyPairsMaps := func(toAppend *topologyPairsMaps) {
-		lock.Lock()
-		defer lock.Unlock()
-		topologyMaps.appendMaps(toAppend)
-	}
-	catchError := func(err error) {
-		lock.Lock()
-		defer lock.Unlock()
-		if firstError == nil {
-			firstError = err
-		}
-	}
+	allNodesCount := len(allNodeNames)
+	errCh := make(chan error, allNodesCount)
+	topologyPairsCh := make(chan *topologyPairsMaps, allNodesCount)
 
 	ctx, cancel := context.WithCancel(context.Background())
 
 	processNode := func(i int) {
 		nodeInfo := nodeInfoMap[allNodeNames[i]]
 		node := nodeInfo.Node()
 		if node == nil {
-			catchError(fmt.Errorf("node not found"))
+			errCh <- fmt.Errorf("node not found")
 			return
 		}
 		for _, existingPod := range nodeInfo.PodsWithAffinity() {
 			existingPodTopologyMaps, err := getMatchingAntiAffinityTopologyPairsOfPod(pod, existingPod, node)
 			if err != nil {
-				catchError(err)
+				errCh <- err
 				cancel()
 				return
 			}
-			appendTopologyPairsMaps(existingPodTopologyMaps)
+			topologyPairsCh <- existingPodTopologyMaps
+		}
+	}
+
+	workqueue.ParallelizeUntil(ctx, 16, allNodesCount, processNode)
+
+	close(topologyPairsCh)
+
+	select {
+	case err := <-errCh:
+		return nil, err
+
```

Comment:
nit: remove extra line.

Reply:
I prefer to make the code into pieces in the select statement. I don't think this would be a problem. But if many folks think it is, I could remove this line.

Reply:
I don't think it is consistent in the codebase now. 

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_attacher.go#L181-L215

```go
		select {
		case event, ok := <-ch:
			if !ok {
				klog.Errorf("[attachment.ID=%v] watch channel had been closed", attachID)
				return "", errors.New("volume attachment watch channel had been closed")
			}

			switch event.Type {
			case watch.Added, watch.Modified:
				attach, _ := event.Object.(*storage.VolumeAttachment)
				successful, err := verifyAttachmentStatus(attach, volumeHandle)
				if err != nil {
					return "", err
				}
				if successful {
					return attachID, nil
				}
			case watch.Deleted:
				// if deleted, fail fast
				klog.Error(log("VolumeAttachment [%s] has been deleted, will not continue to wait for attachment", attachID))
				return "", errors.New("volume attachment has been deleted")

			case watch.Error:
				// close the watcher to avoid keeping the watcher too log
				watcher.Stop()
				watcherClosed = true
				// start another cycle
				return c.waitForVolumeAttachmentInternal(volumeHandle, attachID, timer, timeout)
			}

		case <-timer.C:
			klog.Error(log("attacher.WaitForAttach timeout after %v [volume=%v; attachment.ID=%v]", timeout, volumeHandle, attachID))
			return "", fmt.Errorf("attachment timeout for volume %v", volumeHandle)
		}
```

Reply:
An example does not defeat the convention, scanning through the repo I see a clear pattern in preferring to compress the code...

Reply:
> An example does not defeat the convention, scanning through the repo I see a clear pattern in preferring to compress the code...

Sure, please point out which new lines you prefer to remove.

Reply:
Thanks, note that the compiler has nothing to do with what we are discussing here. By compressing the code, I am only talking about being able to see more lines of code in a single screen when reading/reviewing code.